### PR TITLE
Dev/checkpoint base on fullnode

### PR DESCRIPTION
**Pull request by LonDong-TopNetwork**

### CODE REVIEW

File: src/xtopcom/xblockstore/src/xcheckpoint.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace top {
+namespace store {
+
+struct xtop_checkpoint_cmp {
+    bool operator()(const uint64_t & lhs, const uint64_t & rhs) const {
+        return lhs < rhs;
+    }
+};
+using xcheckpoint_cmp_t = xtop_checkpoint_cmp;
```
Comment (PaytonWu) [inline src/xtopcom/xblockstore/src/xcheckpoint.h:20]: useless?

File: src/xtopcom/xblockstore/src/xcheckpoint.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#include "xbase/xutl.h"
+#if defined(CHECKPOINT_TEST)
+#    include "xblockstore/xcheckpoint_data/xcheckpoint_test.h"
+#elif defined(XBUILD_CI) || defined(XBUILD_DEV)
+#    include "xblockstore/xcheckpoint_data/xcheckpoint_default.h"
+#elif defined(XBUILD_GALILEO)
+#    include "xblockstore/xcheckpoint_data/xcheckpoint_galileo.h"
+#else
+#    include "xblockstore/xcheckpoint_data/xcheckpoint_new_horizons.h"
+#endif
```
Comment (PaytonWu) [inline src/xtopcom/xblockstore/src/xcheckpoint.cpp:19]: move these xcheckpoint_xxx.h into xcheckpoint_data/xcheckpoints.h

**Review by PaytonWu — CHANGES_REQUESTED** [review]

File: src/xtopcom/xblockstore/src/xcheckpoint.h (hunk 1 above)
Comment (LonDong-TopNetwork) [reply src/xtopcom/xblockstore/src/xcheckpoint.h]: already remove useless.

File: src/xtopcom/xblockstore/src/xcheckpoint.cpp (hunk 2 above)
Comment (LonDong-TopNetwork) [reply src/xtopcom/xblockstore/src/xcheckpoint.cpp]: finish gethering those into one file. how about put all "string literals" files in one module, such as genesis info, reset data and checkpoint data.

**Review by PaytonWu — APPROVED** [review]

**Pull request merged**